src/services/address.service.ts

```diff
@@ -35,36 +36,211 @@ class AddressService {
-    //     const toRadians = (degrees: string) => {
-    //         return degrees * (Math.PI / 180);
-    //     }
+    private async getKilometerDistance(lat1: number, lon1: number, lat2: number, lon2: number) {
+        // Defining this function inside of this private method means it's
+        // not accessible outside of it, which is perfect for encapsulation.
+        const toRadians = (degrees: number) => {
+            return degrees * (Math.PI / 180);
+        }
+
+        // Radius of the Earth in KM
+        const R = 6371;
+
+        // Convert Lat and Longs to Radians
+        const dLat = toRadians(lat2 - lat1);
+        const dLon = toRadians(lon2 - lon1);
 
-    //     // Radius of the Earth in KM
-    //     const R = 6371;
+        // Haversine Formula to calculate the distance between two locations
+        // on a sphere.
+        const a =
+            Math.sin(dLat / 2) * Math.sin(dLat / 2) +
+             Math.cos(toRadians(lat1)) * Math.cos(toRadians(lat2)) *
+            Math.sin(dLon / 2) * Math.sin(dLon / 2);
+
+        const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+             // convert and return distance in KM
+            return (R * c);
+    }
+
+    private async getMilesDistance(lat1: number, lon1: number, lat2: number, lon2: number) {
+        const kilometers = await this.getKilometerDistance(lat1, lon1, lat2, lon2);
+        const milesPerKilometer = 0.621371;
+        return (kilometers * milesPerKilometer);
+    }
 
-    //     // Convert Lat and Longs to Radians
-    //     const dLat = toRadians(lat2 - lat1);
-    //     const dLon = toRadians(lon2 - lon1);
+    private async calculateAverageCoordinates(location: any) {
+        if (!location || !Array.isArray(location) || location.length === 0) {
+            const errMessage = "Invalid location data. Please provide a valid array of locations.";
+            loggerService.error({ path: "/address/calculateAverageCoordinates", message: errMessage }).flush();
+            throw new Error(errMessage);
+        }
 
-    //     // Haversine Formula to calculate the distance between two locations
-    //     // on a sphere.
-    //     const a =
-    //         Math.sin(dLat / 2) * Math.sin(dLat / 2) +
-    //         Math.cos(toRadians(lat1)) * Math.cos(toRadians(lat2)) *
-    //         Math.sin(dLon / 2) * Math.sin(dLon / 2);
+        const totalCoordinates = location.reduce(
+            (totals, loc) => {
+            const latitude = parseFloat(loc.latitude);
+            const longitude = parseFloat(loc.longitude);
 
-    //     const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+            if (isNaN(latitude) || isNaN(longitude)) {
+                const errMessage = "Invalid latitude or longitude in location data.";
+                loggerService.error({ path: "/address/calculateAverageCoordinates", message: errMessage }).flush();
+                throw new Error(errMessage);
```

Comment:
Catching this above is decent. I'd argue the issue here is an error for the client and not this app, though. So log as warning.